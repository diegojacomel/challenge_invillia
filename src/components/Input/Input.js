/** Modules */
import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const FieldStyle = styled.input`
    border: 1px solid ${props => props.theme.colors.gray};
    border-radius: ${props => props.theme.spacing.sm};
    background: ${props => props.theme.colors.white};
    padding: ${props => props.theme.doubleSpacing.lg};
    margin-top: ${props => props.theme.spacing.sm};
    width: calc(${props => props.theme.spacing.full} - ${props => props.theme.spacing.llg});
    outline: none;
`

const InputElement = ({ onChange, type, placeholder, id }) => (
    <FieldStyle onChange={onChange} type={type} placeholder={placeholder} id={id} />
)

function Input({ onChange, type, placeholder, id, ...rest }) {
    return (
        <InputElement onChange={onChange} type={type} placeholder={placeholder} {...rest} className="testInput" id={id} />
    )
}

Input.defaultProps = {
    onChange: () => { },
    type: 'text',
    placeholder: 'Digite aqui seu texto',
}

Input.propTypes = {
    onChange: PropTypes.func,
    type: PropTypes.string,
    placeholder: PropTypes.string,
    id: PropTypes.number
}

export default Input;