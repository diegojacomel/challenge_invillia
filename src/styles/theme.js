const primaryTheme = {
    colors: {
        primary: '#007bff',
        success: '#28a745',
        secondary: '#6c757d',
        warning: "#ffc107",
        black: '#000000',
        white: '#FFFFFF',
        gray: '#cccccc',
        grayLight: '#F2F2F2',
        grayDark: '#333333'
    },
    fontSize: {
        xs: "14px",
        sm: "16px",
        md: "20px",
        lg: "25px",
        hg: "40px",
        hhg: "60px"
    },
    spacing: {
        none: '0',
        xxs: '3px',
        xs: '5px',
        ss: '8px',
        sm: '10px',
        ssm: '15px',
        md: '20px',
        mmd: '25px',
        lg: '30px',
        llg: '35px',
        xg: '40px',
        xxg: '50px',
        gt: '80px',
        ggt: '100px',
        hg: '200px',
        hhg: '400px',
        quarter: '25%',
        middle: '50%',
        full: '100%'
    },
    doubleSpacing: {
        xs: "2px 5px",
        sm: "3px 10px",
        md: "5px 12px",
        lg: "8px 16px",
        xg: "10px 20px",
        hg: "20px 40px"
    },
    rounded: {
        none: '0',
        xs: '2px',
        sm: '3px',
        md: '5px',
        lg: '10px'
    }
}

export {
    primaryTheme
}