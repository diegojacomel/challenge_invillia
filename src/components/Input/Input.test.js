// Modules
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';

// Theme
import { primaryTheme } from '../../styles/theme';

// Component
import Input from './Input';

describe('Input', () => {
    const wrapper = shallow(<Input />);

    test('Render without crashing', () => {
        shallow(<Input />);
    })

    test('It exist', () => {
        expect(wrapper.find('.testInput')).toExist();
    })

    test('At least one', () => {
        expect(wrapper).toContainMatchingElement('.testInput');
    })

    test('Have display name', () => {
        expect(wrapper.find('.testInput')).toHaveDisplayName('InputElement');
    })

    test('Match selector', () => {
        expect(wrapper.find('.testInput')).toMatchSelector('InputElement');
    })

    test('Have props', () => {
        expect(wrapper.find('.testInput')).toHaveProp('type');
        expect(wrapper.find('.testInput')).toHaveProp('placeholder');
    })

    it('Render correctly', () => {
        const tree = renderer.create(
            <ThemeProvider theme={primaryTheme}>
                <Input value="value" onChange={() => {}} type="text" />
            </ThemeProvider>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    })
})