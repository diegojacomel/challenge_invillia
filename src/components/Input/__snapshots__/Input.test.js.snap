// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Render correctly 1`] = `
<input
  className="sc-bdVaJa kouTes"
  onChange={[Function]}
  placeholder="Digite aqui seu texto"
  type="text"
/>
`;
